zebra/zebra_rib.c

```diff
@@ -2144,43 +2152,53 @@ static wq_item_status meta_queue_process(struct work_queue *dummy, void *data)
 	return mq->size ? WQ_REQUEUE : WQ_SUCCESS;
 }
 
-/* Look into the RN and queue it into one or more priority queues,
- * increasing the size for each data push done.
+
+/*
+ * Look into the RN and queue it into the highest priority queue
+ * at this point in time for processing.
+ *
+ * We want the Route Node to be enqueued one time.
```

Comment:
This comment I didn't quite get. The route_node still can be poked more than once, can't it? this change will only enqueue to a single subqueue at each invocation - but the route_node could still end up on more than one subqueue if it gets poked more than once. Is the intent to really just be in a single subqueue at most, ever?

Reply:
that is not quite what it does, we loop over all route nodes and find the `lowest` meta queue index value and use that one.

Reply:
talked w/ mark about this on slack and I've cleaned up the comments here